tests/test_mux.py

```diff
@@ -40,45 +46,307 @@ class TestMuxTunnelBase(object):
+
+        fvs = {"vlanid": "1000"}
+        confdb.create_entry("VLAN", "Vlan1000", fvs)
+
+        fvs = {"tagging_mode": "untagged"}
+        confdb.create_entry("VLAN_MEMBER", "Vlan1000|Ethernet0", fvs)
+        confdb.create_entry("VLAN_MEMBER", "Vlan1000|Ethernet4", fvs)
+
+        fvs = {"NULL": "NULL"}
+        confdb.create_entry("VLAN_INTERFACE", "Vlan1000", fvs)
+        confdb.create_entry("VLAN_INTERFACE", "Vlan1000|192.168.0.1/24", fvs)
+
+        dvs.runcmd("config interface startup Ethernet0")
+        dvs.runcmd("config interface startup Ethernet4")
+
+
+    def create_mux_cable(self, confdb):
+
+        fvs = { "server_ipv4":self.SERV1_IPV4+self.IPV4_MASK, "server_ipv6":self.SERV1_IPV6+self.IPV6_MASK }
+        confdb.create_entry(self.CONFIG_MUX_CABLE, "Ethernet0", fvs)
+
+        fvs = { "server_ipv4":self.SERV2_IPV4+self.IPV4_MASK, "server_ipv6":self.SERV2_IPV6+self.IPV6_MASK }
+        confdb.create_entry(self.CONFIG_MUX_CABLE, "Ethernet4", fvs)
+
+
+    def set_mux_state(self, appdb, ifname, state_change):
+
+        ps = swsscommon.ProducerStateTable(appdb, self.APP_MUX_CABLE)
+
+        fvs = create_fvs(state=state_change)
+
+        ps.set(ifname, fvs)
+
+        time.sleep(1)
+
+
+    def check_neigh_in_asic_db(self, asicdb, ip, expected=1):
+
+        nbr = asicdb.wait_for_n_keys(self.ASIC_NEIGH_TABLE, expected)
+
+        found = False
+        for key in nbr:
+            entry = json.loads(key)
+            if entry["ip"] == ip:
+                found = True
+                entry = key
+                break
+
+        assert found
+        return entry
+
+
+    def check_tnl_nexthop_in_asic_db(self, asicdb, expected=1):
+
+        global tunnel_nh_id
+
+        nh = asicdb.wait_for_n_keys(self.ASIC_NEXTHOP_TABLE, expected)
+
+        for key in nh:
+            fvs = asicdb.get_entry(self.ASIC_NEXTHOP_TABLE, key)
+            if fvs.get("SAI_NEXT_HOP_ATTR_TYPE") == "SAI_NEXT_HOP_TYPE_TUNNEL_ENCAP":
+                tunnel_nh_id = key
+
+        assert tunnel_nh_id
+
+
+    def check_nexthop_in_asic_db(self, asicdb, key, standby=False):
+
+        global tunnel_nh_id
+
+        fvs = asicdb.get_entry(self.ASIC_ROUTE_TABLE, key)
+        if not fvs:
+            assert False
+
+        nhid = fvs.get("SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID")
+        if standby:
+            assert (nhid == tunnel_nh_id)
+        else:
+            assert (nhid != tunnel_nh_id)
+
+
+    def check_nexthop_group_in_asic_db(self, asicdb, key, num_tnl_nh=0):
+
+        global tunnel_nh_id
+
+        fvs = asicdb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", key)
+
+        nhg_id = fvs["SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID"]
+
+        fvs = asicdb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP", nhg_id)
+
+        assert bool(fvs)
+
+        keys = asicdb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER")
+
+        # Two NH group members are expected to be added
+        assert len(keys) == 2
```

Comment:
Same with these two lines and `wait_for_n_keys`

Reply:
done